clients/fides-js/src/lib/gpp/stub.ts

```diff
@@ -0,0 +1,171 @@
+/**
+ * Typescript adaptation of https://github.com/IABTechLab/iabgpp-es/blob/master/modules/stub
+ * Refactored to share code with the TCF version
```

Comment:
the original code looks like it was adapted from the TCF version but with some different naming conventions, and not quite as many guardrails as the TCF one has (and not typescript). This standardizes the two library stubs, uses TS, and hopefully has the same guardrails.

Reply:
The general format of these stubs is:

1. Look to see if a CMP already exists on the page
2. If not, instantiate the function (`__gpp` or `__tcfapi`) on the window object. Set up the "generic" commands (ping, addEventListener, etc. these are defined by the spec) to return something (likely that the CMP isn't ready yet)
3. Also set up so that we can `postMessage` (how iframes talk to one another)